src/py-opentimelineio/opentimelineio/console/otiotool.py

```diff
@@ -861,8 +870,29 @@ def summarize_timeline(list_tracks, list_clips, list_media, verify_media,
             if list_tracks:
                 print(f"TRACK: {child.name} ({child.kind})")
         if isinstance(child, otio.schema.Clip):
-            if list_clips:
-                print("  CLIP:", child.name)
+            if list_clips or verify_ranges:
+                if verify_ranges:
+                    range_msg = ""
+                    try:
+                        source = child.source_range
+                        available = child.available_range()
+
+                        src_start = source.start_time.value
+                        src_dur = source.duration.value
+                        avail_start = available.start_time.value
+                        avail_dur = available.duration.value
+
+                        starts_early = src_start < avail_start
+                        ends_late = (src_dur - src_start) > (avail_dur - avail_start)
+                        if starts_early or ends_late:
+                            range_msg = "SOURCE MEDIA OUT OF BOUNDS"
+                        else:
+                            range_msg = "IN BOUNDS"
+                    except Exception:  # available range is, well, unavailable
+                        pass
```

Comment:
So the error condition is that both available and source ranges are present but the source range is not within the available range?

Reply:
Correct, if there's no available range defined than it won't do the bounds check.